### DISCUSSION TITLE
Proposal: Fullscreen only container layout

### DISCUSSION THREAD

**Discussion by theherk** [nikitabobko/AeroSpace#629] [feature-ideas]
With respect to [this comment](https://github.com/nikitabobko/AeroSpace/issues/558#issuecomment-2392490834) in #558, I would like to propose a configuration option.

For starters, thank you for your feedback @nikitabobko, and I agree about the 0 padding being somewhat illogical usually, however there is a use case that may warrant such a configuration. Currently, the `fullscreen` command almost does what is needed, except that it applies to an application rather than a workspace.

The scenario is this: there is at least one application ([Zed](https://zed.dev)), which sort of hides the presence of multiple windows. So if you open multiple projects, there are separate windows that occupy the same coordinates, and when you switch to the application, you get the most recent project. Then, the most practical way to switch projects is to do so in the application's project switcher, which then takes you to the correct window. All this works perfectly fine with the accordion, but since the project switching is done in the application, the accordion padding is just wasting space that isn't providing any practical information.

What would be good is either to be able to set padding per workspace, or have a fullscreen layout for a workspace which makes every application take the full area. That does diminish the information about how many applications are open on a workspace, but in a way that is actually beneficial in certain cases.

---

Also, I completely understand if this is rejected out of hand for adding complexity or ambiguity, but I'm curious if others run into this same thing. I am super happy with the current system. Thank you for all the hard work.

**Comment by liamwh** [comment]
+1, would also love to see a fullscreen only and half-page only layouts.
